### DISCUSSION TITLE
Color mapped

### DISCUSSION THREAD

**Discussion by bimintelligence** [IfcOpenShell/IfcOpenShell#3168] [Q&A]
Hi everyone,

IfcIndexedColourMap defined the assignment of colour information to individual faces.

But in the modelling software, a lot of color definition is depended on their vertex value.
So it represents a gradient value, such as a tessellated surface.

So, is there any more suitable entities to define this kind of case.

BR
Shawn

**Comment by aothms** [comment]
> IfcIndexedColourMap defined the assignment of colour information to individual faces.

I remember discussing this earlier. I think schema-wise it could be quite easily adapted for example by making IfcIndexedColourMap.ColourIndex optional. If not-set, IfcIndexedColourMap.Colours would directly provide vertex colours to IfcTessellatedFaceSet.Coordinates 

Feel free to bring up this discussion here: https://github.com/buildingSMART/IFC4.3.x-development

I personally would propose to postpone this to a proper rework of colours/materials/textures in IFC5. We currently have three overlapping concepts Styled Material - SurfaceStyle on Representation Item - Direct colours on tesselations - (Plus the deprecated Pset_Draughting?). Also, physical materials are rarely interpolated, so this is really a presentation thing, maybe it should just be a texture?

**Reply by bimintelligence** [reply]
Hi aothms, is it means the color will be assigned to the vertexes of the surface.
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/73410751/ffdcbc12-44dc-4f2f-91ca-1255d0f71584)

It defines like below, could you provide an example .ifc file to show this case.
I'm a little confused.

If this entity doesn't work, is there any other entities to solve this case?

BR
Shawn
